src/python/WMCore/Services/RucioConMon/RucioConMon.py

```diff
@@ -59,9 +67,16 @@ def _getResult(self, uri, callname="", clearCache=False, args=None):
 
         if clearCache:
             self.clearCache(cachedApi, args)
-        data = self.refreshCache(cachedApi, apiUrl)
-        results = data.read()
-        data.close()
+        results = '{}' # explicitly define results which will be loaded by json.loads below
+        if binary:
+            with self.refreshCache(cachedApi, apiUrl, decoder=False, binary=True) as istream:
+                results = gzip.decompress(istream.read())
```

Comment:
Valentin, let me suggest another minor improvement here. Can you move L79 (return results) to the end of this "if binary"? Something like:
```
        if binary:
            with self.refreshCache(cachedApi, apiUrl, decoder=False, binary=True) as istream:
                results = gzip.decompress(istream.read())
            return results
```
then you can remove the 2 lines a bit further below.